ldes-server-retention/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/services/retentionpolicy/creation/RetentionPolicyFactoryImpl.java

```diff
@@ -18,22 +17,35 @@
 import java.util.Optional;
 
 import static be.vlaanderen.informatievlaanderen.ldes.server.domain.constants.RdfConstants.RDF_SYNTAX_TYPE;
-import static be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.creation.RetentionPolicyConstants.*;
+import static be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.creation.RetentionPolicyConstants.TIME_BASED_RETENTION_POLICY;
+import static be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.creation.RetentionPolicyConstants.VERSION_BASED_RETENTION_POLICY;
 
 @Component
 public class RetentionPolicyFactoryImpl implements RetentionPolicyFactory {
 
 	private final Map<String, RetentionPolicyCreator> retentionPolicyCreatorMap;
 
-	public RetentionPolicyFactoryImpl(MemberPropertiesRepository memberPropertiesRepository) {
+	public RetentionPolicyFactoryImpl() {
 		this.retentionPolicyCreatorMap = Map.of(
 				TIME_BASED_RETENTION_POLICY, new TimeBasedRetentionPolicyCreator(),
-				VERSION_BASED_RETENTION_POLICY, new VersionBasedRetentionPolicyCreator(memberPropertiesRepository),
-				POINT_IN_TIME_RETENTION_POLICY, new PointInTimeRetentionPolicyCreator());
+				VERSION_BASED_RETENTION_POLICY, new VersionBasedRetentionPolicyCreator()
+		);
 	}
 
 	@Override
-	public List<RetentionPolicy> getRetentionPolicyListForView(ViewSpecification viewSpecification) {
+	public Optional<RetentionPolicy> extractRetentionPolicy(ViewSpecification viewSpecification) {
+		List<RetentionPolicy> policies = getRetentionPolicyListForView(viewSpecification);
+
+		return switch (policies.size()) {
+			case 0 -> Optional.empty();
+			case 1 -> Optional.of(policies.get(0));
+			case 2 -> Optional.of(TimeAndVersionBasedRetentionPolicy.from(policies.get(0), policies.get(1)));
+			default -> throw new IllegalArgumentException("A view cannot have more than 2 retention policies!");
```

Comment:
did not know that was the case ?

Reply:
With PIT removed, we only have timebased and versionbased